clients/python-wrapper/tests/integration/test_branch.py

```diff
@@ -82,3 +83,65 @@ def validate_uncommitted_changes(branch, expected, change_type="added", prefix="
         count += 1
     if count != len(expected):
         raise AssertionError(f"Expected {len(expected)} changes, got {count}")
+
+
+def test_transaction(setup_repo):
+    _, repo = setup_repo
+    path_and_data1 = ["a", "b", "bar/a", "bar/b", "bar/c", "c"]
+    path_and_data2 = ["foo/a", "foo/b", "foo/c"]
+    test_branch = repo.branch("main")
+
+    with test_branch.transact(commit_message="my transaction", commit_metadata={"foo": "bar"}) as tx:
+        assert tx.get_commit().id == test_branch.head.id
+        upload_data(tx, path_and_data1)
+        upload_data(tx, path_and_data2)
+        tx.reset_changes(path_type="common_prefix", path="foo")
+        tx_id = tx.id
+
+    # Verify transaction branch was deleted
+    with expect_exception_context(NotFoundException):
+        repo.branch(tx.id).get_commit()
+
+    #  Verify transaction completed successfully
+    log = list(test_branch.log(amount=2))
+    assert log[0].message == f"Merge transaction {tx_id} to branch"
+    assert log[1].message == "my transaction"
+    assert log[1].metadata.get("foo") == "bar"
+
+    for obj in path_and_data1:
+        assert test_branch.object(obj).exists()
+
+    for obj in path_and_data2:
+        assert not test_branch.object(obj).exists()
+
+    # Reset all changes - ensure no new commits
+    with test_branch.transact(commit_message="my transaction", commit_metadata={"foo": "bar"}) as tx:
+        assert tx.get_commit().id == test_branch.head.id
+        upload_data(tx, path_and_data2)
+        tx.reset_changes()
+
+    log = list(test_branch.log(amount=1))
+    assert log[0].message == f"Merge transaction {tx_id} to branch"
+
+
+def test_transaction_failure(setup_repo):
+    _, repo = setup_repo
+    new_data = ["a", "b", "bar/a", "bar/b"]
+    common_data = ["foo/a", "foo/b", "foo/c"]
+    test_branch = repo.branch("main")
+
+    # Merge on dirty branch
+    with expect_exception_context(TransactionException, "dirty branch"):
+        with test_branch.transact(commit_message="my transaction") as tx:
+            assert tx.get_commit().id == test_branch.head.id
+            upload_data(tx, new_data)
+            upload_data(test_branch, common_data, 2)
+
+    # Verify tx branch was not deleted
+    repo.branch(tx.id).get_commit()
```

Comment:
What is expected here, if a transaction fails, shouldn't clean up its own mess?

Reply:
We intentionally decided to leave the branch undeleted for debugging purposes